testing/trino-tests/src/test/java/io/trino/sql/planner/BaseCostBasedPlanTest.java

```diff
@@ -221,15 +221,14 @@ protected Path getSourcePath()
     {
         Path workingDir = Paths.get(System.getProperty("user.dir"));
         verify(isDirectory(workingDir), "Working directory is not a directory");
-        String topDirectoryName = workingDir.getFileName().toString();
-        switch (topDirectoryName) {
-            case "trino-tests":
-                return workingDir;
-            case "trino":
-                return workingDir.resolve("testing/trino-tests");
-            default:
-                throw new IllegalStateException("This class must be executed from trino-tests or Trino source directory");
+        if (isDirectory(workingDir.resolve(".git"))) {
```

Comment:
I wonder if we should check for existence of `.gitignore` file instead since that is part of the checked out source.

Reply:
you can have `.gitignore` at any level within the repo, but `.git` folder should be only one